scripts/install.js

```diff
@@ -323,14 +317,30 @@ if (process.env.SENTRYCLI_SKIP_DOWNLOAD === '1') {
   process.exit(0);
 }
 
-(async () => {
-  try {
-    await downloadBinary();
-    await checkVersion();
-    process.exit(0);
-  } catch (e) {
-    // eslint-disable-next-line no-console
-    console.error(e.toString());
-    process.exit(1);
-  }
-})();
+const { packageName: distributionPackageName, subpath: distributionSubpath } =
+  helper.getDistributionForThisPlatform();
+
+if (distributionPackageName === undefined) {
+  helper.throwUnsupportedPlatformError();
+}
+
+try {
+  require.resolve(`${distributionPackageName}/${distributionSubpath}`);
+  // If the `resolve` call succeeds it means a binary was installed successfully via optional dependencies so we can skip the manual postinstall download.
+  process.exit(0);
+} catch (e) {
+  // Optional dependencies likely didn't get installed - proceed with fallback downloading manually
+  logger.log(
+    `Sentry CLI binary installation via optional dependencies was unsuccessful. Downloading manually instead.`
```

Comment:
Should we at least log the error here?

Reply:
The error would just be Node's default "Cannot resolve module `@sentry/cli-whatever-architecture`" which is not helpful at all here - so likely not. During runtime (which is the actually important part because we cannot control what happens between installation and runtime - people do random shit like moving the platform specific binary from Windows to Debian in Docker) we will print a more specific error message with better instructions.